lib/theory/src/Theory/Tools/InjectiveFactInstances.hs

```diff
@@ -9,63 +11,189 @@
 -- Computate an under-approximation to the set of all facts with unique
 -- instances, i.e., fact whose instances never occur more than once in a
 -- state. We use this information to reason about protocols that exploit
--- exclusivity of linear facts.
+-- exclusivity of facts.
 module Theory.Tools.InjectiveFactInstances (
 
   -- * Computing injective fact instances.
-  simpleInjectiveFactInstances
+  MonotonicBehaviour(..)
+  , simpleInjectiveFactInstances
   ) where
 
 import           Extension.Prelude   (sortednub)
 
 -- import           Control.Applicative
 import           Control.Monad.Fresh
-import           Data.Label
+import           Control.DeepSeq
+
+import           GHC.Generics        (Generic)
+import           Data.Label          as L
 import qualified Data.Set            as S
+import qualified Data.Map            as M
+import           Data.List
+import           Data.Maybe
+import           Data.Binary
 import           Safe                (headMay)
+import           Control.Applicative (empty)
+import           Debug.Trace
 
 import           Theory.Model
+--import Control.DeepSeq (NFData)
+
+-- unspecified = there is no rule using this fact
+-- unstable = increasing and decreasing or not at all related inputs and outputs
+data MonotonicBehaviour = Constant | Increasing | Decreasing | StrictlyIncreasing | StrictlyDecreasing | Unstable | Unspecified
```

Comment:
Is it possible the the changes in this file are part of the development of that paper "Subterm-based proof techniques for improving the automation and scope of security protocol analysis" and arrived here accidentally?

Reply:
Indeed, this code comes from the Subterms. It is unused code that facilitates merging with #507. When doing this merge, then one has to only process all `--TODO-Subterm when enabling subterms ...`.